lib/hls/hls_parser.js

```diff
@@ -1220,6 +1221,17 @@ shaka.hls.HlsParser = class {
 
       const videoRange = tag.getAttributeValue('VIDEO-RANGE');
 
+      // According to the HLS spec:
+      // By default a video variant is monoscopic, so an attribute
+      // consisting entirely of REQ-VIDEO-LAYOUT="CH-MONO" is unnecessary
+      // and SHOULD NOT be present.
+      let videoLayout = tag.getAttributeValue('REQ-VIDEO-LAYOUT') || 'CH-MONO';
+      if (videoLayout == 'CH-STEREO,CH-MONO') {
```

Comment:
Also from the spec:

> Clients should ignore any unrecognized enumerated-strings in the
 specifier list.

So I think we are expected to ignore any unknown video layout strings in this list?
Presumably as future-proofing, so that if they add more layout strings in a later spec older clients won't stop working.
Given that, we could have an implementation more like...

```js
let videoLayout = tag.getAttributeValue('REQ-VIDEO-LAYOUT');
if (videoLayout && videoLayout.includes(',')) {
  // If multiple video layout strings are present, pick the first valid one.
  const layoutStrings = videoLayout.split(',').filter((layoutString) => {
    return layoutString == 'CH-STEREO' || layoutString == 'CH-MONO';
  });
  videoLayout = layoutStrings[0];
}
// According to the HLS spec:
// By default a video variant is monoscopic, so an attribute
// consisting entirely of REQ-VIDEO-LAYOUT="CH-MONO" is unnecessary
// and SHOULD NOT be present.
videoLayout ||= 'CH-MONO';
```

Reply:
Changed, with some linter changes

Reply:
The linter doesn't accept it :(